lnbits/static/js/components/extension-settings.js

```diff
@@ -0,0 +1,113 @@
+Vue.component('lnbits-extension-settings-form', {
+  name: 'lnbits-extension-settings-form',
+  props: ['options', 'name', 'cancel'],
+  data: function () {
+    return {
+      settings: undefined,
+      usr: undefined,
+      admin: false
+    }
+  },
+  methods: {
+    endpoint: function () {
+      return `/${this.name}/api/v1/settings?usr=${this.usr}`
+    },
+    updateSettings: async function () {
+      try {
+        const {data} = await LNbits.api.request(
+          'PUT',
+          this.endpoint(),
+          null,
+          this.settings
+        )
+        this.settings = data
+      } catch (error) {
+        LNbits.utils.notifyApiError(error)
+      }
+    },
+    getSettings: async function () {
+      try {
+        const {data} = await LNbits.api.request('GET', this.endpoint())
+        this.settings = data
+      } catch (error) {
+        LNbits.utils.notifyApiError(error)
+      }
+    },
+    resetSettings: async function () {
+      try {
+        await LNbits.api.request('DELETE', this.endpoint())
+        await this.getSettings()
+      } catch (error) {
+        LNbits.utils.notifyApiError(error)
+      }
+    }
+  },
+  created: async function () {
+    if (window.user.admin) {
+      this.admin = true
+      this.usr = window.user.id
```

Comment:
- NOK to access global state from components, better to pass as a property
- should apply `principle of least privilege`, that is pass an `adminkey` instead of the full user id